src/Data/PostObjectMutation.php

```diff
@@ -355,6 +355,13 @@ protected static function set_object_terms( int $post_id, array $input, WP_Post_
 						return;
 					}
 
+					if ( 'category' === $tax_object->name  ) {
```

Comment:
Does the bug happen also when `$append` is false? If not, it would be great to gate against and avoid the unnecessary extra db write.